hibernate-core/src/main/java/org/hibernate/boot/model/source/spi/AbstractAttributeKey.java

```diff
@@ -138,11 +130,11 @@ public boolean isCollectionElement() {
 	 * marker ({@link #COLLECTION_ELEMENT}.
 	 */
 	public boolean isPartOfCollectionElement() {
-		return fullPath.contains( DOT_COLLECTION_ELEMENT );
+		return fullPath.contains( getDelimiter() + COLLECTION_ELEMENT );
 	}
 
 	public String stripCollectionElementMarker() {
-		return DOT_COLLECTION_ELEMENT_PATTERN.matcher( fullPath ).replaceAll( Matcher.quoteReplacement( "" ) );
+		return fullPath.replace( getDelimiter() + COLLECTION_ELEMENT, "" );
```

Comment:
the original RegExp usage is to squeeze the perf gain to the last drop, but it seems these two methods are seldom used; if we still aim that, we might design passing the demiter in constructor or builder, then we could still compile a literal Pattern from it.